src/Common/src/Interop/Unix/libcurl/Interop.libcurl_types.cs

```diff
@@ -166,8 +175,41 @@ internal struct curl_version_info_data
             private int versionNum;
             private unsafe char *host;
             internal int features;
+            internal IntPtr ssl_version;
+        }
+
+        internal static partial class CurlSslBackend
+        {
+            /// Currently libcurl does not reliably return the backend code from
+            ///        curl_easy_getinfo(easy, CURLINFO_TLS_SESSION, &x);
+            /// for all backends. 
+            /// Based on the advice at curl-library mailing list, till they fix it,
+            /// we will use the curl_version_info_data.ssl_version
+
+            internal static bool AllowsSslCallback(IntPtr callback)
+            {
+                // we can actually check the name against the backend
+                //but, there are other SSL backends that also support callbacks - libreSSL, cyassl/wolfSSL etc
+                // they achieve so by providing an openssl-compatible API
+
+                // we actually try and set the option on a dummy handle with the given pointer
+                // and verify that the return code is CURLE_OK
+                SafeCurlHandle easy = Interop.libcurl.curl_easy_init();
+                int code = Interop.libcurl.curl_easy_setopt(easy, CURLoption.CURLOPT_SSL_CTX_FUNCTION, callback);
```

Comment:
Don't just pass garbage pointers in to native code... make a static delegate to an empty function to use as the test.

If libcurl were to try and invoke it... ::panics::